support/ebpf/go_labels.ebpf.c

```diff
@@ -0,0 +1,266 @@
+// This file contains the code for extracting custom labels from Go runtime.
+
+#include "bpfdefs.h"
+#include "kernel.h"
+#include "tracemgmt.h"
+#include "tsd.h"
+#include "types.h"
+#include "util.h"
+
+static inline __attribute__((__always_inline__)) void
+process_value(GoMapBucket *map_value, CustomLabelsArray *out, unsigned i)
+{
+  if (map_value->tophash[i] == 0)
+    return;
+  if (out->len >= MAX_CUSTOM_LABELS)
+    return;
+  CustomLabel *lbl = &out->labels[out->len];
+  if (map_value->keys[i].str != NULL) {
+    unsigned klen = MIN(map_value->keys[i].len, CUSTOM_LABEL_MAX_KEY_LEN - 1);
+    long res      = bpf_probe_read_user(lbl->key, klen, map_value->keys[i].str);
+    if (res) {
```

Comment:
nit: we generally have just tested for error from `bpf_probe_read_user` directly in the `if` and not printed the result value. I suppose there is an error code. Do you think it adds value to print error code here? Would simplify the code a bit if `res` was not kept...